### PR TITLE
Adding notes & team properties.  Redesign versions.

### PR DESCRIPTION
**Pull request by JeffAshton**
As per slack discussion:

https://d2l.slack.com/archives/C12B7Q13N/p1626350893026400

### CODE REVIEW

File: schemas/farm-multivariate/v5_0.json [hunk 1]
```diff
@@ -8,14 +8,16 @@
 		"name",
```
Comment (JeffAshton) [inline schemas/farm-multivariate/v5_0.json:8]: Lather, rinse, repeat...

File: schemas/farm-boolean/v5_0.json [hunk 2]
```diff
@@ -7,14 +7,16 @@
 		"$schema",
 		"name",
 		"environments",
-		"rallyId"
+		"rallyId",
+		"team",
+		"versions"
```
Comment (JeffAshton) [inline schemas/farm-boolean/v5_0.json:12]: `team` and `versions` become required properties

File: schemas/farm-boolean/v5_0.json [hunk 3]
```diff
@@ -27,9 +29,40 @@
...
+				"type": "string",
+				"minLength": 1
+			},
+			"minItems": 1,
+			"title": "A non structured list of notes about the feature"
+		},
+		"team": {
+			"type": "string",
+			"pattern": "^[a-z_]+$",
+			"title": "The name of the team that owns the feature"
```
Comment (JeffAshton) [inline schemas/farm-boolean/v5_0.json:44]: Will get mapped to a teams json file in the repo which will contain slack channel to reach out to.

File: schemas/farm-boolean/v5_0.json [hunk 4]
```diff
@@ -27,9 +29,40 @@
 			"title": "The description of what the feature flag controls",
 			"minLength": 1
 		},
-		"removedFromCode": {
```
Comment (JeffAshton) [inline schemas/farm-boolean/v5_0.json:30]: Replaced by `versions/removed`

File: schemas/instance-boolean/v5_0.json [hunk 5]
```diff
@@ -7,14 +7,16 @@
 		"$schema",
```
Comment (JeffAshton) [inline schemas/instance-boolean/v5_0.json:7]: Lather, rinse, repeat...

File: schemas/instance-multivariate/v5_0.json [hunk 6]
```diff
@@ -8,14 +8,16 @@
 		"name",
```
Comment (JeffAshton) [inline schemas/instance-multivariate/v5_0.json:8]: Lather, rinse, repeat...

File: schemas/org-boolean/v5_0.json [hunk 7]
```diff
@@ -7,14 +7,16 @@
 		"$schema",
```
Comment (JeffAshton) [inline schemas/org-boolean/v5_0.json:7]: Lather, rinse, repeat...

File: schemas/org-multivariate/v5_0.json [hunk 8]
```diff
@@ -8,14 +8,16 @@
 		"name",
```
Comment (JeffAshton) [inline schemas/org-multivariate/v5_0.json:8]: Lather, rinse, repeat...

File: test/testCases/endToEnd/farm-boolean-v5_0/definition.json [hunk 9]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/farm-boolean-v5_0/definition.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/farm-multivariate-v5_0/definition.json [hunk 10]
```diff
@@ -0,0 +1,85 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/farm-multivariate-v5_0/definition.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/farm-multivariate-v5_0/expected.json [hunk 11]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/farm-multivariate-v5_0/expected.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/instance-boolean-v5_0/definition.json [hunk 12]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/instance-boolean-v5_0/definition.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/instance-multivariate-v5_0/definition.json [hunk 13]
```diff
@@ -0,0 +1,97 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/instance-multivariate-v5_0/definition.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/org-boolean-v5_0/definition.json [hunk 14]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/org-boolean-v5_0/definition.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/org-multivariate-v5_0/definition.json [hunk 15]
```diff
@@ -0,0 +1,116 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/org-multivariate-v5_0/definition.json:1]: Test with team, notes, & versions

File: test/testCases/endToEnd/farm-boolean-v5_0/expected.json [hunk 16]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/farm-boolean-v5_0/expected.json:1]: The output from the input file above that Launch Darkly use to get

File: test/testCases/endToEnd/farm-multivariate-v5_0/expected.json (hunk 11 above)
Comment (JeffAshton) [inline test/testCases/endToEnd/farm-multivariate-v5_0/expected.json:1]: The output from the input file above that Launch Darkly use to get

File: test/testCases/endToEnd/instance-boolean-v5_0/expected.json [hunk 17]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/instance-boolean-v5_0/expected.json:1]: The output from the input file above that Launch Darkly use to get

File: test/testCases/endToEnd/instance-multivariate-v5_0/expected.json [hunk 18]
```diff
@@ -0,0 +1,117 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/instance-multivariate-v5_0/expected.json:1]: The output from the input file above that Launch Darkly use to get

File: test/testCases/endToEnd/org-boolean-v5_0/expected.json [hunk 19]
```diff
@@ -0,0 +1,120 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/org-boolean-v5_0/expected.json:1]: The output from the input file above that Launch Darkly use to get

File: test/testCases/endToEnd/org-multivariate-v5_0/expected.json [hunk 20]
```diff
@@ -0,0 +1,145 @@
+{
```
Comment (JeffAshton) [inline test/testCases/endToEnd/org-multivariate-v5_0/expected.json:1]: The output from the input file above that Launch Darkly use to get

**Review by dlockhart — APPROVED** [review]
This will make my tracking spreadsheet redundant, which is awesome. In @semibran's new UI, I think the team is being fetched from Rally based on the US/DE associated with the flag... so that info may be redundant. But also not coupling to Rally's APIs is nice.

**Comment by JeffAshton** [comment]
> This will make my tracking spreadsheet redundant, which is awesome. In @semibran's new UI, I think the team is being fetched from Rally based on the US/DE associated with the flag... so that info may be redundant. But also not coupling to Rally's APIs is nice.

Team is driven more by my own desires for automation within the repo of pinging teams about stuff.  Well I could definitely call out to rally, not sure it's a dependency I really want.

**Review by njostonehouse — APPROVED** [review]

**Comment by njostonehouse** [comment]
I think Team probably makes more sense than a Rally ID, since flags likely cross multiple Rally objects more often than they cross teams.

**Review by elainejonesd2l — APPROVED** [review]

**Comment by wallace-c** [comment]
> I think Team probably makes more sense than a Rally ID, since flags likely cross multiple Rally objects more often than they cross teams.

@njostonehouse RallyID is a requirement from the release management team to be able to associate features to work, so unfortunately will stay :)

**Review by wallace-c — APPROVED** [review]

**Pull request merged**